akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/NullBlogState.java

```diff
@@ -0,0 +1,181 @@
+  }
+  public static class AddPostDone implements BlogCommand {
+    final String postId;
+
+    public AddPostDone(String postId) {
+      this.postId = postId;
+    }
+  }
+  public static class GetPost implements BlogCommand {
+    final ActorRef<PostContent> replyTo;
+
+    public GetPost(ActorRef<PostContent> replyTo) {
+      this.replyTo = replyTo;
+    }
+  }
+  public static class ChangeBody implements BlogCommand {
+    final String newBody;
+    final ActorRef<Done> replyTo;
+
+    public ChangeBody(String newBody, ActorRef<Done> replyTo) {
+      this.newBody = newBody;
+      this.replyTo = replyTo;
+    }
+  }
+  public static class Publish implements BlogCommand {
+    final ActorRef<Done> replyTo;
+
+    public Publish(ActorRef<Done> replyTo) {
+      this.replyTo = replyTo;
+    }
+  }
+  public static class PassivatePost implements BlogCommand {
+
+  }
+  public static class PostContent implements BlogCommand {
+    final String postId;
+    final String title;
+    final String body;
+
+    public PostContent(String postId, String title, String body) {
+      this.postId = postId;
+      this.title = title;
+      this.body = body;
+    }
+  }
+
+  public static class BlogBehavior extends PersistentBehavior<BlogCommand, BlogEvent, BlogState> {
+
+    private CommandHandlerBuilder<BlogCommand, BlogEvent, BlogState, BlogState> initialCommandHandler() {
+      return commandHandlerBuilder(Objects::isNull)
+          .matchCommand(AddPost.class, cmd -> {
+            PostAdded event = new PostAdded(cmd.content.postId, cmd.content);
+            return Effect().persist(event)
+                .andThen(() -> cmd.replyTo.tell(new AddPostDone(cmd.content.postId)));
+          })
+          .matchCommand(PassivatePost.class, (state, cmd) -> Effect().stop());
+    }
+
+    private CommandHandlerBuilder<BlogCommand, BlogEvent, BlogState, BlogState> postCommandHandler() {
+      return commandHandlerBuilder(Objects::nonNull)
+          .matchCommand(ChangeBody.class, (state, cmd) -> {
+            BodyChanged event = new BodyChanged(state.postId(), cmd.newBody);
+            return Effect().persist(event).andThen(() -> cmd.replyTo.tell(Done.getInstance()));
+          })
+          .matchCommand(Publish.class, (state, cmd) -> Effect()
+              .persist(new Published(state.postId())).andThen(() -> {
+                System.out.println("Blog post published: " + state.postId());
+                cmd.replyTo.tell(Done.getInstance());
+              }))
+          .matchCommand(GetPost.class, (state, cmd) -> {
+            cmd.replyTo.tell(state.postContent);
+            return Effect().none();
+          })
+          .matchCommand(AddPost.class, (state, cmd) -> Effect().unhandled())
+          .matchCommand(PassivatePost.class, (state, cmd) -> Effect().stop());
+    }
+
+    public BlogBehavior(PersistenceId persistenceId) {
+      super(persistenceId);
+    }
+
+    @Override
+    public BlogState emptyState() {
+      return null;
```

Comment:
Could it default to `null` in the superclass?

Reply:
I'd rather make people think about it than we deciding that you always use `null`. There are many cases where there is a natural zero, and others where it's better to use a EmptyState class if you anyway have several ADT style classes for the state lifecycle.

Reply:
I didn't mean always `null`, just by default. I think it's a reasonable default for most Java developers, who might not be used to modelling empty state in terms of ADTs. It would reduce the number of decisions someone has to make when starting out. You'll always be able to override it if you want to. What's the benefit of forcing someone to think about this up front?

Reply:
If it's not an abstract method they will not see that it exists. It's also good to make the conscious decision that it null to be aware of that when the first command arrives before saving the first event the `state` parameter in the command handler will be null.